silverstripe/silverstripe-framework
# FIX: Defer loading of UploadField EditForm iframes (fixes #2711)

Tested in:
- Win XP: IE8
- Win 7: IE9 & 10
- Win 8: IE10 (desktop)
- OSX: Chrome, Firefox & Safari.

Implements the fix suggested issue #1439:
- Prevent loading by default (change <iframe src... to <iframe data-src, and use that attr for dynamic loading)
- Show loading indicator when edit button is pressed, until iframe is loaded
- Ensure it doesn't break when edit button is pressed multiple times during iframe load
- Ensure that heights are correctly calculated on iframe load

I don’t like the binding `iframe.on('load')` events in the `onclick` handler, but entwine doesn’t support event binding on iframes (apparently).

Before load:
![screen shot 2013-12-12 at 12 59 35](https://f.cloud.github.com/assets/1655548/1733131/04f9befc-6330-11e3-8004-768dfe8aca49.png)

During load:
![screen shot 2013-12-12 at 13 12 14](https://f.cloud.github.com/assets/1655548/1733133/0ae8c7f4-6330-11e3-8db9-325754dc7a21.png)

IE8 (ignore text highlighting):
![screen shot 2013-12-12 at 14 50 16](https://f.cloud.github.com/assets/1655548/1734681/ef1db39e-6349-11e3-94b7-ce48e4ed3bbe.png)
